File: public/docs/_examples/dependency-injection/dart/lib/heroes/heroes_component.dart
// #docregion
import 'package:angular2/core.dart';

import 'hero_list_component.dart';
import 'hero_service_provider.dart';

@Component(
    selector: 'my-heroes',
    template: '''
      <h2>Heroes</h2>
      <hero-list></hero-list>''',
    providers: const [heroServiceProvider],
    directives: const [HeroListComponent])
class HeroesComponent {}

File: public/docs/_examples/dependency-injection/ts/app/heroes/heroes.component.ts
// #docregion
import { Component }          from '@angular/core';

import { HeroListComponent }  from './hero-list.component';
import { heroServiceProvider } from './hero.service.provider';

@Component({
  selector: 'my-heroes',
  template: `
  <h2>Heroes</h2>
  <hero-list></hero-list>
  `,
  providers: [heroServiceProvider],
  directives: [HeroListComponent]
})
export class HeroesComponent { }
